crates/forge/src/lib.rs

```diff
@@ -285,23 +289,25 @@ fn try_close_tmp_dir(maybe_tmp_dir: Option<TempDir>) -> Result<()> {
 /// * `contracts` - Map with names of contract used in tests and corresponding sierra and casm artifacts
 /// * `predeployed_contracts` - Absolute path to predeployed contracts used by starknet state e.g. account contracts
 ///
-#[allow(clippy::implicit_hasher)]
-pub fn run(
-    package_root: &Utf8PathBuf,
+
+#[allow(clippy::implicit_hasher, clippy::too_many_arguments)]
+pub async fn run(
+    package_root: Arc<Utf8PathBuf>,
```

Comment:
I'm not sure if it makes sense to use `Arc` for all possible types. It is more efficient than copying complex structs like `RunnerParams` etc. but for simpler stuff, I'd just use normal references (`&`) and clone where necessary.

Reply:
We should generally avoid keeping stuff on the heap if not necessary